### PR TITLE
refactor(common): Reduce the precision to 2 digits in the ngOptimized…

### LINKED ISSUE #50273: Aspect ratios are not rounded in the NgOptimizedImage warnings
### Which @angular/* package(s) are the source of the bug?

common

### Is this a regression?

No

### Description

When the warning is displayed because the aspect ratios are not matching correctly the numbers are not rounded and thus show quite a few decimal places

image

### Please provide a link to a minimal reproduction of the bug

_No response_

### Please provide the exception or error you saw

_No response_

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI 16.0.01
```

### Anything else?

Created at the behest of @Andre

### PR DESCRIPTION
**Pull request by JeanMeche**
…Image distortion warning message

fixes #50273

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [x] No

### CODE REVIEW

**Review by AndrewKushnir — COMMENTED** [review]
@JeanMeche thanks for the PR, the change looks great 👍 

There are some CI failures, where we'd probably need to update the `packages/core/test/bundling/image-directive/e2e/image-distortion/image-distortion.e2e-spec.ts` to use shorter version of aspect ratio values.

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts [hunk 1]
```diff
@@ -1012,3 +1017,8 @@ function assertNoLoaderParamsWithoutLoader(dir: NgOptimizedImage, imageLoader: I
             `To fix this, provide a custom loader or remove the \`loaderParams\` attribute from the image.`));
   }
 }
+
+
+function round2Decimals(numb: number): number {
+  return Math.round((numb + Number.EPSILON) * 100) / 100;
+}
```
Comment (AndrewKushnir) [inline packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:1024]: Quick question: why not keeping the `.toFixed(2)` as you had before?
If that was needed to keep integers untouched (avoid adding `.00`), we can do something like this:
```suggestion
function round(input: number): number {
  return Number.isInteger(input) ? input : input.toFixed(2);
}
```

Comment (JeanMeche) [reply packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:1024]: Yeah that was it, thanks !

**Comment by AndrewKushnir** [comment]
@JeanMeche thanks for addressing the feedback, the change looks great 👍 Could you please take a look at the CI failures when you get a chance? I suspect one of the spec files require an update.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:532261911:BASE:532261934:1684193228289:1b19f06).

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit cd86eb4606657c5d972b73a57f6a6e58edb7b0b9.

**Pull request closed without merging**